src/PKSim.BatchTool/Services/BatchLogger.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Events;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Container;
+using OSPSuite.Utility.Events;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+
+namespace PKSim.BatchTool.Services
+{
+   public class BatchLogger : ILogger
+   {
+      private readonly IEventPublisher _eventPublisher;
+      public string Name { get; }
+      private readonly LogLevel _logLevel;
+
+      public BatchLogger(string name, LogLevel logLevel)
+      {
+         _eventPublisher = IoC.Resolve<IEventPublisher>();
+         _logLevel = logLevel;
+         Name = name;
+      }
+
+      public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+      {
+         if (!IsEnabled(logLevel))
+            return;
+
+         if (formatter == null)
+            throw new ArgumentNullException(nameof(formatter));
+
+         string message = formatter(state, exception);
+         if (string.IsNullOrEmpty(message) && exception == null)
+            return;
+
+         writeMessage(logLevel, Name, eventId.Id, message, exception);
+      }
+
+      private void writeMessage(LogLevel logLevel, string name, int eventIdId, string message, Exception exception)
+      {
+         var messageStatus = notificationTypeFrom(logLevel);
+         var logEntry = new LogEntry(messageStatus, message);
+         _eventPublisher.PublishEvent(new LogEntryEvent(logEntry));
```

Comment:
That's the code we had before. Now encapsulated and started from the app that needs it